docs/tutorials/bson-v4.txt

```diff
@@ -588,14 +588,30 @@ all time calculations using integer math, as inexactness of floating point
 calculations may produce unexpected results.
 
 
-Date And DateTime Instances
----------------------------
+DateTime Instances
+------------------
 
 BSON only supports storing the time as the number of seconds since the
-Unix epoch. Ruby's ``Date`` and ``DateTime`` instances can be serialized
-to BSON, but when the BSON is deserialized the times will be returned as
+Unix epoch. Ruby's ``DateTime`` instances can be serialized to BSON,
+but when the BSON is deserialized the times will be returned as
 ``Time`` instances.
 
+``DateTime`` class in Ruby supports non-Gregorian calendars. When non-Gregorian
+``DateTime`` instances are serialized, they are first converted to Gregorian
+calendar, and the respective date in the Gregorian calendar is stored in the
+database.
+
+
+Date Instances
+--------------
+
+BSON only supports storing the time as the number of seconds since the
+Unix epoch. Ruby's ``Date`` instances can be serialized to BSON, but when
+the BSON is deserialized the times will be returned as ``Time`` instances.
+
+When ``Date`` instances are serialized, the time value used is the midnight
```

Comment:
I think would say "midnight" instead of "the midnight" but that's not a blocker.

Reply:
:+1: